test/e2e/web-socket-server-and-url.test.js

```diff
@@ -454,6 +454,144 @@ for (const webSocketServerType of webSocketServerTypes) {
     });
   });
 
+  describe('should work when "port" option is "auto"', () => {
+    beforeAll((done) => {
+      const options = {
+        webSocketServer: webSocketServerType,
+        port: 'auto',
+        host: '0.0.0.0',
+        client: {
+          webSocketURL: {
+            port: port2,
+          },
+        },
+      };
+      testServer.startAwaitingCompilation(config, options, done);
+    });
+
+    afterAll(testServer.close);
+
+    describe('browser client', () => {
+      it('should work', (done) => {
+        runBrowser().then(({ page, browser }) => {
+          waitForTest(browser, page, /ws/, (websocketUrl) => {
+            expect(websocketUrl).toContain(
+              `${websocketUrlProtocol}://localhost:${port2}/ws`
+            );
+
+            done();
+          });
+
+          page.goto(`http://localhost:8080/main`);
+        });
+      });
+    });
+  });
+
+  describe('should work with the "port" option is 80', () => {
+    beforeAll((done) => {
+      const options = {
+        webSocketServer: webSocketServerType,
+        port: 80,
+        host: '0.0.0.0',
+        client: {
+          webSocketURL: {
+            port: port2,
+          },
```

Comment:
Can we add all tests without `webSocketURL`?, otherwise tests do not have sense

Reply:
I will update.